// react-storefront/search/SearchSuggestions.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = SearchSuggestions;
exports.styles = void 0;

var _react = _interopRequireWildcard(require("react"));

var _styles = require("@material-ui/core/styles");

var _SearchSuggestionGroup = _interopRequireDefault(require("./SearchSuggestionGroup"));

var _SearchContext = _interopRequireDefault(require("./SearchContext"));

var _LoadMask = _interopRequireDefault(require("../LoadMask"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var styles = function styles(theme) {
  return {
    /**
     * Styles applied to the root element.
     */
    root: {
      flex: 1,
      position: 'relative',
      overflowY: 'auto'
    },

    /**
     * Styles applied to each of the group wrapper elements.
     */
    group: {
      margin: theme.spacing(0, 0, 2, 0)
    }
  };
};

exports.styles = styles;
var useStyles = (0, _styles.makeStyles)(styles, {
  name: 'RSFSearchSuggestions'
});

function SearchSuggestions(_ref) {
  var classes = _ref.classes,
      render = _ref.render;
  classes = useStyles({
    classes: classes
  });

  var _useContext = (0, _react.useContext)(_SearchContext["default"]),
      state = _useContext.state;

  return /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, /*#__PURE__*/_react["default"].createElement(_LoadMask["default"], {
    show: state.loading,
    transparent: true
  }), /*#__PURE__*/_react["default"].createElement("div", {
    className: classes.root
  }, render ? render(state) : state.groups && state.groups.map(function (group) {
    return /*#__PURE__*/_react["default"].createElement("div", {
      key: group.caption,
      className: classes.group
    }, /*#__PURE__*/_react["default"].createElement(_SearchSuggestionGroup["default"], group));
  })));
}

SearchSuggestions.propTypes = {
  /**
   * Override or extend the styles applied to the component. See [CSS API](#css) below for more details.
   */
  classes: _propTypes["default"].object
};
SearchSuggestions.defaultProps = {};
