// react-storefront/search/SearchSuggestionGroup.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = SearchSuggestionGroup;
exports.styles = void 0;

var _react = _interopRequireDefault(require("react"));

var _makeStyles = _interopRequireDefault(require("@material-ui/core/styles/makeStyles"));

var _Typography = _interopRequireDefault(require("@material-ui/core/Typography"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _SearchSuggestionItem = _interopRequireDefault(require("./SearchSuggestionItem"));

var styles = function styles(theme) {
  return {
    /**
     * Styles applied to the root element.
     */
    root: {
      listStyle: 'none',
      margin: theme.spacing(2),
      '& a strong': {
        fontWeight: 'bold',
        color: 'inherit'
      }
    },

    /**
     * Styles applied to the group's caption element.
     */
    caption: {
      textTransform: 'uppercase',
      fontWeight: 'bold',
      paddingBottom: 5,
      borderBottom: "1px solid ".concat(theme.palette.divider),
      margin: theme.spacing(0, 0, 1, 0)
    },

    /**
     * Styles applied to the group's list element.
     */
    list: {
      '&[data-ui=list]': {
        padding: 0,
        margin: theme.spacing(0, 0, 4, 0)
      },
      '&[data-ui=thumbnails]': {
        display: 'flex',
        listStyle: 'none',
        margin: '0 -15px',
        padding: '0 10px',
        overflowX: 'auto',
        '& > li': {
          margin: '5px'
        }
      }
    }
  };
};

exports.styles = styles;
var useStyles = (0, _makeStyles["default"])(styles, {
  name: 'RSFSearchSuggestionGroup'
});

function SearchSuggestionGroup(_ref) {
  var classes = _ref.classes,
      ui = _ref.ui,
      caption = _ref.caption,
      links = _ref.links,
      children = _ref.children;
  classes = useStyles({
    classes: classes
  });
  return /*#__PURE__*/_react["default"].createElement("div", {
    className: classes.root
  }, /*#__PURE__*/_react["default"].createElement(_Typography["default"], {
    className: classes.caption
  }, caption), /*#__PURE__*/_react["default"].createElement("ul", {
    className: classes.list,
    "data-ui": ui
  }, children ? children : links.map(function (item, i) {
    return /*#__PURE__*/_react["default"].createElement(_SearchSuggestionItem["default"], {
      item: item,
      ui: ui,
      key: i
    });
  })));
}

SearchSuggestionGroup.propTypes = {
  /**
   * Override or extend the styles applied to the component. See [CSS API](#css) below for more details.
   */
  classes: _propTypes["default"].object,

  /**
   * Child nodes that can be used for the suggestion group instead of mapping the [links](#prop-links).
   */
  children: _propTypes["default"].node,

  /**
   * An array of props to pass to a [SearchSuggestionItem](/apiReference/search/SearchSuggestionItem)
   * with data for suggested searches.
   */
  links: _propTypes["default"].arrayOf(_propTypes["default"].shape({
    as: _propTypes["default"].string,
    href: _propTypes["default"].string,
    text: _propTypes["default"].string,
    pageData: _propTypes["default"].object,
    thumbnail: _propTypes["default"].object
  })),

  /**
   * A title for the list.
   */
  caption: _propTypes["default"].string.isRequired
};
SearchSuggestionGroup.defaultProps = {};
