// react-storefront/utils/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "fetchLatest", {
  enumerable: true,
  get: function get() {
    return _fetchLatest["default"];
  }
});
Object.defineProperty(exports, "format", {
  enumerable: true,
  get: function get() {
    return _format["default"];
  }
});
Object.defineProperty(exports, "getBase64ForImage", {
  enumerable: true,
  get: function get() {
    return _getBase64ForImage["default"];
  }
});
Object.defineProperty(exports, "isBrowser", {
  enumerable: true,
  get: function get() {
    return _isBrowser["default"];
  }
});
Object.defineProperty(exports, "isSameOrigin", {
  enumerable: true,
  get: function get() {
    return _isSameOrigin["default"];
  }
});
Object.defineProperty(exports, "lazyLoadImages", {
  enumerable: true,
  get: function get() {
    return _lazyLoadImages["default"];
  }
});
Object.defineProperty(exports, "merge", {
  enumerable: true,
  get: function get() {
    return _merge["default"];
  }
});
Object.defineProperty(exports, "minifyStyles", {
  enumerable: true,
  get: function get() {
    return _minifyStyles["default"];
  }
});
Object.defineProperty(exports, "mod", {
  enumerable: true,
  get: function get() {
    return _mod["default"];
  }
});
Object.defineProperty(exports, "useDebounce", {
  enumerable: true,
  get: function get() {
    return _useDebounce["default"];
  }
});
Object.defineProperty(exports, "withCaching", {
  enumerable: true,
  get: function get() {
    return _withCaching["default"];
  }
});
Object.defineProperty(exports, "withDefaultHandler", {
  enumerable: true,
  get: function get() {
    return _withDefaultHandler["default"];
  }
});

var _fetchLatest = _interopRequireDefault(require("./fetchLatest.js"));

var _format = _interopRequireDefault(require("./format.js"));

var _getBase64ForImage = _interopRequireDefault(require("./getBase64ForImage.js"));

var _isBrowser = _interopRequireDefault(require("./isBrowser.js"));

var _isSameOrigin = _interopRequireDefault(require("./isSameOrigin.js"));

var _lazyLoadImages = _interopRequireDefault(require("./lazyLoadImages.js"));

var _merge = _interopRequireDefault(require("./merge.js"));

var _minifyStyles = _interopRequireDefault(require("./minifyStyles.js"));

var _mod = _interopRequireDefault(require("./mod.js"));

var _useDebounce = _interopRequireDefault(require("./useDebounce.js"));

var _withCaching = _interopRequireDefault(require("./withCaching.js"));

var _withDefaultHandler = _interopRequireDefault(require("./withDefaultHandler.js"));
