// react-storefront/CartButton.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = CartButton;
exports.styles = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Link = _interopRequireDefault(require("./link/Link"));

var _ToolbarButton = _interopRequireDefault(require("./ToolbarButton"));

var _ShoppingCart = _interopRequireDefault(require("@material-ui/icons/ShoppingCart"));

var _styles = require("@material-ui/core/styles");

var _Badge = _interopRequireDefault(require("@material-ui/core/Badge"));

var _clsx = _interopRequireDefault(require("clsx"));

var _buttonProps;

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { (0, _defineProperty2["default"])(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

var styles = function styles(theme) {
  return {
    /**
     * Styles applied to the [`Link`](/apiReference/link/Link) element used as the root.
     */
    link: {},

    /**
     * Styles passed through to the [`Badge`](https://material-ui.com/api/badge/#css) element's
     * `badge` CSS rule.
     */
    badge: {
      border: '2px solid white',
      padding: '0 4px'
    },

    /**
     * Styles applied to the button icon.
     */
    icon: {
      color: theme.palette.action.active
    }
  };
};

exports.styles = styles;
var useStyles = (0, _styles.makeStyles)(styles, {
  name: 'RSFCartButton'
});
/**
 * A cart header button that display the number of items in the cart using a badge.
 *
 * Example:
 *
 * ```js
 * <CartButton href="/cart" quantity={1} />
 * ```
 */

function CartButton(_ref) {
  var classes = _ref.classes,
      href = _ref.href,
      as = _ref.as,
      onClick = _ref.onClick,
      icon = _ref.icon,
      quantity = _ref.quantity,
      linkProps = _ref.linkProps,
      badgeProps = _ref.badgeProps,
      buttonProps = _ref.buttonProps;
  classes = useStyles({
    classes: classes
  });

  var cartIcon = icon || /*#__PURE__*/_react["default"].createElement(_ShoppingCart["default"], {
    className: classes.icon
  });

  return /*#__PURE__*/_react["default"].createElement(_Link["default"], (0, _extends2["default"])({}, linkProps.anchorProps, {
    className: (0, _clsx["default"])(classes.link, linkProps.className),
    href: href,
    as: as,
    onClick: onClick
  }), /*#__PURE__*/_react["default"].createElement(_ToolbarButton["default"], buttonProps, /*#__PURE__*/_react["default"].createElement(_Badge["default"], (0, _extends2["default"])({
    badgeContent: quantity
  }, badgeProps, {
    classes: _objectSpread(_objectSpread({}, badgeProps.classes), {}, {
      badge: classes.badge
    })
  }), cartIcon)));
}

CartButton.propTypes = {
  /**
   * The url path to the cart page.
   */
  href: _propTypes["default"].string,

  /**
   * Override or extend the styles applied to the component. See [CSS API](#css) below for more details.
   */
  classes: _propTypes["default"].object,

  /**
   * Optional custom cart icon.
   */
  icon: _propTypes["default"].node,

  /**
   * Props passed through to the [`ToolbarButton`](/apiReference/ToolbarButton#props) element.
   */
  buttonProps: _propTypes["default"].object,

  /**
   * Props passed through to the [`Badge`](https://material-ui.com/api/badge/#props) element.
   */
  badgeProps: _propTypes["default"].object,

  /**
   * Props passed through to the [`Link`](/apiReference/link/Link#props) element.
   */
  linkProps: _propTypes["default"].object
};
CartButton.defaultProps = {
  href: '/cart',
  badgeProps: {
    color: 'primary'
  },
  buttonProps: (_buttonProps = {}, (0, _defineProperty2["default"])(_buttonProps, 'aria-label', 'Cart'), (0, _defineProperty2["default"])(_buttonProps, "color", 'inherit'), _buttonProps),
  linkProps: {
    anchorProps: {
      'data-th': 'cart-link'
    }
  }
};
