// react-storefront/Label.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = Label;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireDefault(require("react"));

var _clsx = _interopRequireDefault(require("clsx"));

var _Typography = _interopRequireDefault(require("@material-ui/core/Typography"));

var _styles = require("@material-ui/core/styles");

var styles = function styles(theme) {
  return {
    /**
     * Styles applied to the root element\.
     */
    root: {
      fontWeight: 500,
      marginRight: 10
    }
  };
};

var useStyles = (0, _styles.makeStyles)(styles, {
  name: 'RSFLabel'
});

function Label(_ref) {
  var classes = _ref.classes,
      className = _ref.className,
      props = (0, _objectWithoutProperties2["default"])(_ref, ["classes", "className"]);
  classes = useStyles({
    classes: classes
  });
  return /*#__PURE__*/_react["default"].createElement(_Typography["default"], (0, _extends2["default"])({}, props, {
    className: (0, _clsx["default"])(className, classes.root)
  }));
}

Label.propTypes = {
  /**
   * Override or extend the styles applied to the component. See [CSS API](#css) below for more details.
   */
  classes: _propTypes["default"].object,

  /**
   * CSS class to apply to the root element
   */
  className: _propTypes["default"].string
};
