// react-storefront/Offline.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = Offline;

var _react = _interopRequireDefault(require("react"));

var _styles = require("@material-ui/core/styles");

var _Typography = _interopRequireDefault(require("@material-ui/core/Typography"));

var _CloudOff = _interopRequireDefault(require("@material-ui/icons/CloudOff"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var styles = function styles(theme) {
  return {
    /**
     * Styles applied to the root element.
     */
    root: {
      display: 'flex',
      flexDirection: 'column',
      width: '100%',
      alignItems: 'center',
      marginTop: '40px',
      color: '#999'
    },

    /**
     * Styles applied to the icon element.
     */
    icon: {
      fontSize: 60,
      color: '#999'
    },

    /**
     * Styles applied to the heading element.
     */
    heading: {},

    /**
     * Styles applied to the message element.
     */
    message: {}
  };
};

var useStyles = (0, _styles.makeStyles)(styles, {
  name: 'RSFOffline'
});
/**
 * A page to display when in Offline mode
 */

function Offline(_ref) {
  var classes = _ref.classes,
      heading = _ref.heading,
      message = _ref.message,
      Icon = _ref.Icon;
  classes = useStyles({
    classes: classes
  });
  return /*#__PURE__*/_react["default"].createElement("div", {
    className: classes.root
  }, /*#__PURE__*/_react["default"].createElement(Icon, {
    className: classes.icon
  }), /*#__PURE__*/_react["default"].createElement(_Typography["default"], {
    variant: "h6",
    component: "h1",
    className: classes.heading
  }, heading), /*#__PURE__*/_react["default"].createElement(_Typography["default"], {
    variant: "caption",
    className: classes.message
  }, message));
}

Offline.propTypes = {
  /**
   * Override or extend the styles applied to the component. See [CSS API](#css) below for more details.
   */
  classes: _propTypes["default"].object,

  /**
   * Text (or an element) to display as the heading.
   */
  heading: _propTypes["default"].oneOfType([_propTypes["default"].string, _propTypes["default"].element]),

  /**
   * Text (or an element) to display as the message.
   */
  message: _propTypes["default"].oneOfType([_propTypes["default"].string, _propTypes["default"].element]),

  /**
   * An icon to display.
   */
  Icon: _propTypes["default"].elementType
};
Offline.defaultProps = {
  heading: "You're offline",
  message: 'Please check your internet connection',
  Icon: function Icon() {
    return /*#__PURE__*/_react["default"].createElement(_CloudOff["default"], null);
  }
};
